plugin-server/src/main/ingestion-queues/batch-processing/each-batch-webhooks.ts

```diff
@@ -69,7 +69,9 @@ export async function eachBatchWebhooksHandlers(
     const transaction = Sentry.startTransaction({ name: `eachBatchWebhooks` })
 
     try {
-        const batchesWithOffsets = groupIntoBatchesWebhooks(batch.messages, concurrency, actionMatcher)
+        const batchesWithOffsets = groupIntoBatchesByUsage(batch.messages, concurrency, (teamId) =>
+            actionMatcher.hasWebhooks(teamId)
+        )
```

Comment:
nit: you might be able to get away with passing the function ref itself? Would need to confirm it compiles fine
```suggestion
        const batchesWithOffsets = groupIntoBatchesByUsage(batch.messages, concurrency, actionMatcher.hasWebhooks)
```